code/modules/integrated_electronics/subtypes/lists.dm

```diff
@@ -102,14 +115,15 @@
 
 /obj/item/integrated_circuit/lists/at
 	name = "at circuit"
-	desc = "This circuit will pick an element from a list by the input index."
+	desc = "This circuit will pick an element from a list by the input index. Will also output value if item is associated with a value."
 	extended_desc = "If there is no element with such index, result will be null."
 	inputs = list(
 		"list" = IC_PINTYPE_LIST,
 		"index" = IC_PINTYPE_INDEX
 		)
 	outputs = list(
-		"item" = IC_PINTYPE_ANY
+		"item" = IC_PINTYPE_ANY,
```

Comment:
index is numerical. if they want to use a key they can put it in the key section of delete. as for write, you append a key/value pair if it's a new pair and otherwise you can just set it.1

Reply:
for del it already supports using a key for deletion instead of a numerical index
for write i've added a new set circuit for writing values specifically.

Reply:
But why? I can't get reason.

Reply:
Hm actually  I'll think about it soonish

Reply:
I just realized what you mean hold on